soc/arm64/ti_sitara/am6x/mmu_regions.c

```diff
@@ -8,6 +8,9 @@
 #include <zephyr/devicetree.h>
 #include <zephyr/sys/util.h>
 
+/* Only 4KB granularity is supported */
+#define ROUND_UP_MMU_PAGE(x) ROUND_UP((x), CONFIG_MMU_PAGE_SIZE)
```

Comment:
This is a bit gross...

The pinctrl registers are only 0x2ac (684) bytes.
The MMU minimum granularity is 4kB.

The arm64 MMU driver seems to ignore this configuration unless it is rounded up to a page size.
I'm not sure if this is the correct fix, or if this "rounding up" should be done in the MMU macros or driver code..

Reply:
it's really odd having a MMIO space that is not 4k aligned. What else is in that page besides pinctlr?

Reply:
The PADCFG register block is actually 32kB:
![image](https://user-images.githubusercontent.com/124088381/221964700-09b6b508-c117-4952-acb5-c3d8465ca634.png)
Although only a small section of this block is used for pinctrl.

Here's a dtsi from Linux:
https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/ti/k3-am62-main.dtsi#L187

Reply:
After converting to use MMIO this is no longer an issue.
`device_map` calls `k_mem_region_align` which rounds up to an MMU page size